zScanner/SupportFiles/Extensions/String.swift

```diff
@@ -41,4 +41,16 @@ extension String {
         return String(self[start ..< end])
     }
     
+    /* if the personal identification number (PIN) a has length equal to 9 so year must be 53 and less othewise length has to be 10 */
+    func isPIN() -> Bool {
+        if let number = Int(self) {
+            
+            if (number/10000000) <= 53 && length == 9 { return true }
```

Comment:
Discussed elsevere. Use more obvious validating logic.

Reply:
It makes more sense now hopefully.